cpp/CMakeLists.txt

```diff
@@ -615,7 +615,7 @@ endif(UNIX)
 set(ARROW_LINK_LIBS)
 
 # Libraries to link statically with libarrow.so
-set(ARROW_STATIC_LINK_LIBS double-conversion)
+set(ARROW_STATIC_LINK_LIBS ${DOUBLE_CONVERSION_TARGET})
```

Comment:
If we keep the indirection, this should be something like `${DOUBLE_CONVERSION_LIBRARY}`.